tests/testsuite/rustflags.rs

```diff
@@ -690,6 +731,26 @@ fn build_rustflags_normal_source_with_target() {
         .with_status(101)
         .with_stderr_contains("[..]bogus[..]")
         .run();
+
+    // With -Ztarget-applies-to-host build.rustflags should still not apply to build.rs, even when
+    // target == host, to match legacy Cargo behavior.
+    p.change_file("build.rs", "fn main() {}");
+    p.change_file(
+        ".cargo/config",
+        r#"
+        target-applies-to-host = true
+
+        [build]
+        rustflags = ["-Z", "bogus"]
+        "#,
+    );
+    p.cargo("build --lib --target")
+        .arg(host)
+        .masquerade_as_nightly_cargo()
+        .arg("-Ztarget-applies-to-host")
+        .with_status(101)
+        .with_stderr_does_not_contain("[..]build_script_build[..]")
```

Comment:
I'm not really sure what this test is looking for since this is such a generic assertion. I suspect that Cargo is internally failing to run `rustc --version` with the new `-Zbogus` flag and it's not getting anywhere near compiling the build script. Is that the case and if so can this test be updated?

Reply:
I'm removing the modification of this test since it's fully encapsulated by the `build_rustflags_for_build_scripts` test case (which I'm also updating to be more specific in what it tests).